gap/examples.gi

```diff
@@ -370,3 +370,889 @@ GeneralisedPetersenGraphCons);
+  fi;
+  D := Digraph(IsMutableDigraph, []);
+  DigraphAddVertices(D, n);
+  DigraphAddEdges(D, List([1 .. (n - 1)], x -> [x, x + 1]));
+  return DigraphSymmetricClosure(D);
+end);
+
+InstallMethod(PathGraphCons,
+"for IsImmutableDigraph, integer",
+[IsImmutableDigraph, IsInt],
+function(filt, n)
+  local D;
+  D := MakeImmutable(PathGraphCons(IsMutableDigraph, n));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
+  SetIsEmptyDigraph(D, n = 1);
+  return D;
+end);
+
+InstallMethod(PathGraph, "for an integer", [IsInt],
+n -> PathGraphCons(IsImmutableDigraph, n));
+
+InstallMethod(PathGraph, "for a function and an integer",
+[IsFunction, IsInt], PathGraphCons);
+
+InstallMethod(StackedPrismGraphCons, "for IsMutableDigraph and two integers",
+[IsMutableDigraph, IsInt, IsInt],
+function(filt, n, k)
+  if (not IsInt(n)) or (not IsInt(k)) or n < 3 or k < 1 then
+    ErrorNoReturn("the arguments <n> and <k> must be integers, ",
+                  "with <n> greater than 2 and <k> greater than 0,");
+  fi;
+  return DigraphCartesianProduct(CycleGraph(IsMutableDigraph, n),
+          PathGraph(IsMutableDigraph, k));
+end);
+
+InstallMethod(StackedPrismGraphCons,
+"for IsImmutableDigraph, integer, integer",
+[IsImmutableDigraph, IsInt, IsInt],
+function(filt, n, k)
+  local D;
+  D := MakeImmutable(StackedPrismGraphCons(IsMutableDigraph, n, k));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
+  return D;
+end);
+
+InstallMethod(StackedPrismGraph, "for two integers", [IsInt, IsInt],
+{n, k} -> StackedPrismGraphCons(IsImmutableDigraph, n, k));
+
+InstallMethod(StackedPrismGraph, "for a function and two integers",
+[IsFunction, IsInt, IsInt], StackedPrismGraphCons);
+
+InstallMethod(WebGraphCons, "for IsMutableDigraph and an integer",
+[IsMutableDigraph, IsInt],
+function(filt, n)
+  local D, i;
+  if (not IsInt(n)) or n < 3 then
+    ErrorNoReturn("the argument <n> must be an integer greater than 2,");
+  fi;
+  D := StackedPrismGraph(IsMutableDigraph, n, 3);
+  for i in [1 .. (n - 1)] do
+    DigraphRemoveEdge(D, i, i + 1);
+    DigraphRemoveEdge(D, i + 1, i);
+  od;
+  return D;
+end);
+
+InstallMethod(WebGraphCons,
+"for IsImmutableDigraph, integer",
+[IsImmutableDigraph, IsInt],
+function(filt, n)
+  local D;
+  D := MakeImmutable(WebGraphCons(IsMutableDigraph, n));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
+  return D;
+end);
+
+InstallMethod(WebGraph, "for an integer", [IsInt],
+n -> WebGraphCons(IsImmutableDigraph, n));
+
+InstallMethod(WebGraph, "for a function and an integer",
+[IsFunction, IsInt], WebGraphCons);
+
+InstallMethod(HanoiGraphCons, "for IsMutableDigraph and an integer",
+[IsMutableDigraph, IsInt],
+function(filt, n)
+  local D, C, i, nrVert, exp;
+  if (not IsInt(n)) or n < 1 then
+    ErrorNoReturn("the argument <n> must be an integer greater than 0,");
+  fi;
+  C := CycleGraph(IsMutableDigraph, 3);
+  D := C;
+  for i in [2 .. n] do
```

Comment:
Rather than repeatedly creating new digraphs and copying edges over, it would probably be better if you define a single graph with the correct number of nodes and then iteratively construct the graph (using essentially the same method in the loop, something like:

```
DigraphAddEdges(D, Concatenation(DigraphEdges(D) + x, DigraphEdges(D) + (2 * x)));
```

Reply:
> Rather than repeatedly creating new digraphs and copying edges over, it would probably be better if you define a single graph with the correct number of nodes and then iteratively construct the graph (using essentially the same method in the loop, something like:
> 
> ```
> DigraphAddEdges(D, Concatenation(DigraphEdges(D) + x, DigraphEdges(D) + (2 * x)));
> ```

I will get on this, I think you're right.